### PR TITLE
Add github action to build and publish wheels

### PR DESCRIPTION
**Pull request by hardbyte**
Adds a Github Actions workflow to build and optionally publish binary artifacts for major operating systems and Python versions using [`cibuildwheel`](https://github.com/pypa/cibuildwheel).

The action very much follows the example from the cibuildwheel docs - https://cibuildwheel.readthedocs.io/en/stable/setup/#github-actions

The **publish** step only runs when it sees a tag that starts with `v` - then this step will publish all the artifacts to pypi. Note there is already a repository secret in this github repo with a tightly scoped PYPI api token that is allowed to upload packages for just anonlink.

You can see the working Github Action [here](https://github.com/data61/anonlink/actions/runs/1913134745).

I suggest removing the azure build wheel steps - I can do that in this PR if you like?

### CODE REVIEW

File: .github/workflows/build_wheels.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Build and upload to PyPI
+
+on: [push, pull_request]
```
Comment (wilko77) [inline .github/workflows/build_wheels.yml:3]: do we really need to do this on every push? It's about 30 minutes of building wheels and we use the resources for free.
I'm leaning towards building for every pull request and release.
Did you know that github action can be triggered on [release](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release)?

File: .github/workflows/build_wheels.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.3.1
+        # to supply options, put them in 'env', like:
+        env:
+         CIBW_SKIP: '*-win32 *i686'
+
+      - uses: actions/upload-artifact@v2
+        with:
+          path: ./wheelhouse/*.whl
```
Comment (wilko77) [inline .github/workflows/build_wheels.yml:24]: you could give the artifact a more descriptive name.
Currently it just shows up as:
image

**Review by wilko77 — APPROVED** [review]
Nice job.

File: .github/workflows/build_wheels.yml (hunk 1 above)
Comment (hardbyte) [reply .github/workflows/build_wheels.yml:3]: Oh I didn't know that about releases - will add.

Agree, we could be more conscientious of the generous free resources github actions is allowing us to use - will change to run on PR and release.

File: .github/workflows/build_wheels.yml (hunk 2 above)
Comment (hardbyte) [reply .github/workflows/build_wheels.yml:24]: I didn't realize we could download these! I thought it was just a staging area to hand things over between parallel jobs. Will give it a name

**Pull request merged**